:iff -Naur systemd-214/src/libsystemd/sd-bus/test-bus-memfd.c systemd-214.tpg/src/libsystemd/sd-bus/test-bus-memfd.c
--- systemd-214/src/libsystemd/sd-bus/test-bus-memfd.c	2014-03-18 17:05:51.374377792 +0000
+++ systemd-214.tpg/src/libsystemd/sd-bus/test-bus-memfd.c	2014-06-18 23:13:27.910309954 +0000
@@ -151,6 +151,8 @@
         /* check content */
         assert_se(memcmp(buf, "ll", 2) == 0);
 
+#ifndef __UCLIBC__
+       /* uClibc doesn't support pwritev and preadv, skip this check */
         /* writev it out*/
         iov[0].iov_base = (char *)"ABC";
         iov[0].iov_len = 3;
@@ -168,7 +170,7 @@
         iov[2].iov_base = bufv[2];
         iov[2].iov_len = 3;
         assert_se(preadv(fd, iov, 3, 0) == 9);
-
+#endif
         /* check content */
         assert_se(memcmp(bufv[0], "ABC", 3) == 0);
         assert_se(memcmp(bufv[1], "DEF", 3) == 0);
diff -Naur systemd-214/src/shared/missing.h systemd-214.tpg/src/shared/missing.h
--- systemd-214/src/shared/missing.h	2014-05-14 20:58:43.124845369 +0000
+++ systemd-214.tpg/src/shared/missing.h	2014-06-18 23:14:48.741409353 +0000
@@ -29,6 +29,10 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <errno.h>
+#ifdef __UCLIBC__
+/* uClibc has its own sysinfo */
+#define _LINUX_SYSINFO_H 1
+#endif
 #include <linux/oom.h>
 #include <linux/input.h>
 #include <linux/if_link.h>
diff --git a/src/libsystemd/sd-bus/bus-control.c b/src/libsystemd/sd-bus/bus-control.c
index 076bbce..b28d842 100644
--- a/src/libsystemd/sd-bus/bus-control.c
+++ b/src/libsystemd/sd-bus/bus-control.c
@@ -26,6 +26,7 @@
 #include <stddef.h>
 #include <errno.h>
 
+#include "missing.h"
 #include "strv.h"
 #include "sd-bus.h"
 #include "bus-internal.h"
diff --git a/src/libsystemd/sd-bus/sd-bus.c b/src/libsystemd/sd-bus/sd-bus.c
index 1f1a4d3..6ec7b8d 100644
--- a/src/libsystemd/sd-bus/sd-bus.c
+++ b/src/libsystemd/sd-bus/sd-bus.c
@@ -33,7 +33,6 @@
 #include "macro.h"
 #include "strv.h"
 #include "set.h"
-#include "missing.h"
 #include "def.h"
 #include "cgroup-util.h"
 #include "bus-label.h"
diff --git a/src/shared/util.h b/src/shared/util.h
index 0f8c393..638bb01 100644
--- a/src/shared/util.h
+++ b/src/shared/util.h
@@ -42,6 +42,7 @@
 #include <locale.h>
 #include <mntent.h>
 #include <sys/socket.h>
+#include "missing.h"
 
 #if SIZEOF_PID_T == 4
 #  define PID_FMT "%" PRIu32
-- 
1.9.0

